### PR TITLE
Combine benchmarking

### PR DESCRIPTION
**Pull request by johannes-manner**
Combines the platform event executions and the local REST calls together in a single file. 
This is an extention of the performance data utilities from aws and azure to integrate a bechmarking.log file generated during the execution of REST calls.

### CODE REVIEW

**Comment by johannes-manner** [comment]
Fixes the issues #13 and #15.
The changes are tested manually. Especially the model and utility classes are altered and new interfaces were introduced.

File: src/main/java/de/uniba/dsg/serverless/util/BenchmarkingRESTAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+		// read the lines
+		// Sample line: 2018-05-25 13:44:54.826;[pool-2-thread-1];INFO ;de.uniba.dsg.serverless.benchmark.FunctionTrigger;69cfe08c-2b1e-4344-8a2c-0e7732f23146;284d7ba0-b0ce-40cb-9c5b-49696af5ded8;PLATFORMID
+		try {
+			List<String> lines = Files.readAllLines(benchmarkingFile);
+			Predicate<String> isNotEmpty = s -> !s.trim().isEmpty();
+			Function<String, String[]> splitLine = s -> s.split(System.getProperty("CSV_SEPARATOR"));
+			Consumer<String[]> insertEvent = (String[] s) -> {
+				String uuid = s[5].trim();
+				if (temporaryMap.get(uuid) == null) {
```
Comment (martin-endress) [inline src/main/java/de/uniba/dsg/serverless/util/BenchmarkingRESTAnalyzer.java]: containsKey(uuid)

File: src/main/java/de/uniba/dsg/serverless/benchmark/FunctionTrigger.java [hunk 2]
```diff
@@ -59,9 +76,30 @@ public String call() throws SeMoDeException {
...
+			throw new SeMoDeException("Request exited with an error: " + response.getStatus() + " - " + response.getStatusInfo());
+		}
+		
+		// the response entity has to be a json representation with a platformId and a result key
+		String responseEntity = response.readEntity(String.class);
+		String platformId = "";
+		
+		try {
+			JsonNode responseNode = jsonReader.readTree(responseEntity);
+			platformId = responseNode.get(PLATFORM_ID).asText();
```
Comment (martin-endress) [inline src/main/java/de/uniba/dsg/serverless/benchmark/FunctionTrigger.java]: response may not contain PLATFORM_ID and throw a NullPointerException.

File: src/main/java/de/uniba/dsg/serverless/aws/AWSLogHandler.java [hunk 3]
```diff
@@ -278,14 +301,33 @@ private void createDirectoryIfNotExist() throws IOException {
...
 
+	/**
+	 * Filters the log streams, based on the start and end time specified in this class.
+	 * 
+	 * @param logStreams
+	 * @return
+	 */
+	private List<LogStream> filterLogStreams(List<LogStream> logStreams) {
+
+		long startMillis = startTime.toInstant(ZoneOffset.ofTotalSeconds(0)).toEpochMilli();
```
[the remark this answers is not in the document]
Comment (johannes-manner) [reply src/main/java/de/uniba/dsg/serverless/aws/AWSLogHandler.java]: Adressed by #19

**Pull request merged**